test/framework/easyblock.py

```diff
@@ -2172,46 +2172,96 @@ def test_extension_set_start_dir(self):
         cwd = os.getcwd()
         self.assertTrue(os.path.exists(cwd))
 
-        def check_ext_start_dir(expected_start_dir):
+        def check_ext_start_dir(expected_start_dir, unpack_src=True):
             """Check start dir."""
             # make sure we're in an existing directory at the start
             change_dir(cwd)
             eb = EasyBlock(ec['ec'])
             eb.extensions_step(fetch=True, install=False)
             # extract sources of the extension
             ext = eb.ext_instances[-1]
-            ext.run(unpack_src=True)
-            abs_expected_start_dir = os.path.join(eb.builddir, expected_start_dir)
-            self.assertTrue(os.path.samefile(ext.cfg['start_dir'], abs_expected_start_dir))
-            self.assertTrue(os.path.samefile(os.getcwd(), abs_expected_start_dir))
+            ext.run(unpack_src=unpack_src)
+
+            if expected_start_dir is None:
+                self.assertIsNone(ext.start_dir)
+            else:
+                self.assertTrue(os.path.isabs(ext.start_dir))
+                if ext.start_dir != os.sep:
+                    self.assertFalse(ext.start_dir.endswith(os.sep))
+                if os.path.isabs(expected_start_dir):
+                    abs_expected_start_dir = expected_start_dir
+                else:
+                    abs_expected_start_dir = os.path.join(eb.builddir, expected_start_dir)
```

Comment:
@Flamefire as you have taught me, this is not necessary, isn't it?
```suggestion
                abs_expected_start_dir = os.path.join(eb.builddir, expected_start_dir)
```